bakery/src/scripts/checksum-resources.py

```diff
@@ -0,0 +1,137 @@
+    dest_path = os.path.join(output_dir, RESOURCES_DIR)
+    dest_img_filename = os.path.join(dest_path, sha1)
+    rel_raw_img_filename = os.path.relpath(raw_img_filename, dest_path)
+    try:
+        # set a relative symlink to raw file
+        os.symlink(rel_raw_img_filename, dest_img_filename)
+    except FileExistsError:     # ignore symlink file existing
+        pass
+
+
+def create_json_metadata(output_dir, sha1, mime_type, s3_md5, original_name):
+    """ Create json with MIME type of a (symlinked) resource file """
+    data = {}
+    data['original_name'] = original_name
+    data['mime_type'] = mime_type
+    data['s3_md5'] = s3_md5
+    data['sha1'] = sha1
+    json_filename = os.path.join(output_dir, RESOURCES_DIR, sha1+'.json')
+    with open(json_filename, 'w') as outfile:
+        json.dump(data, outfile)
+
+
+def generate_checksum_resources_from_xhtml(filename, output_dir):
+    """ Go through HTML and checksum/copy/mime type all <img> and <a> resources """
+    doc = etree.parse(filename)
+    source_path = os.path.dirname(filename)
+    basename = os.path.basename(filename)
+    module_name = os.path.splitext(basename)[0]
+    # get all img @src resources
+    for node in doc.xpath('//x:img[@src and not(starts-with(@src, "http") or starts-with(@src, "//"))]',
+                          namespaces={'x': 'http://www.w3.org/1999/xhtml'}):
+        img_filename = node.attrib['src']
+        img_filename = os.path.join(source_path, img_filename)
+        img_basename = os.path.basename(img_filename)
+
+        sha1, s3_md5 = get_checksums(img_filename)
+        mime_type = get_mime_type(img_filename)
+
+        if sha1:    # file exists
+            node.attrib['src'] = '../' + RESOURCES_DIR + '/' + sha1
+
+            create_symlink(img_filename, output_dir, sha1)
+            create_json_metadata(output_dir, sha1, mime_type, s3_md5, img_basename)
+            # print('{}, {}, {}'.format(img_filename, sha1, mime_type)) # debug
+
+    # get all a @href resources
+    for node in doc.xpath('//x:a[@href and not(starts-with(@href, "http") or starts-with(@href, "//") or starts-with(@href, "#"))]',
+                          namespaces={'x': 'http://www.w3.org/1999/xhtml'}):
+        img_filename = node.attrib['href']
+
+        # fix a @href links to module_name directories
+        # they are pointing relatively to ./image.jpg but need to point to ./m123/image.jpg
+        img_filename = os.path.join(source_path, module_name, img_filename)
+        img_basename = os.path.basename(img_filename)
+
+        sha1, s3_md5 = get_checksums(img_filename)
+        mime_type = get_mime_type(img_filename)
+
+        if sha1:    # file exists
+            node.attrib['href'] = '../' + RESOURCES_DIR + '/' + sha1
+
+            create_symlink(img_filename, output_dir, sha1)
+            create_json_metadata(output_dir, sha1, mime_type, s3_md5, img_basename)
+            # print('{}, {}, {}'.format(img_filename, sha1, mime_type)) # debug
```

Comment:
Minor cleanup: Did you want to uncomment this for logging purposes or perhaps remove?

Reply:
good point